AR-js-org/AR.js
# Make distance of gps-new-entity-place available as soon as it's added

**⚠️ All PRs have to be done versus 'dev' branch, so be aware of that, or we'll close your issue ⚠️**

**What kind of change does this PR introduce?**


Allows the `distance` property of `gps-new-entity-place`s to be available as soon as they are added.
**Can it be referenced to an Issue? If so what is the issue # ?**

**How can we test it?**



Try out the `show-distance` example from A-Frame `new-location-based` examples. Previously, you had to set the location initially (to create the boxes) and then set it again to access the `distance` property. Now the `distance` property is available as soon as the boxes are added (you need to input a location for them to appear, then click on them to test).

**Summary**


This solves issue #502. It works by reintroducing the `gps-entity-place-added` event, which was available in the older location-based components. The `gps-new-camera` receives this event, and sends its own location back to the `gps-new-entity-place` to set its distance.

**Does this PR introduce a breaking change?**

No

**Please TEST your PR before proposing it. Specify here what device you have used for tests, version of OS and version of Browser**

Firefox latest, Linux, desktop. 

**Other information**
